src/Chapter3.hs

```diff
@@ -971,6 +1068,22 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Enum, Show, Eq, Ord, Bounded)
+
+isWeekend :: Day -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: Day -> Day
+nextDay d
+  | d == maxBound = minBound
```

Comment:
Nice solution! It would even work if you decide to change your weekday implementation to the American system